x-pack/plugins/apm/public/components/app/TransactionOverview/index.tsx

```diff
@@ -38,6 +39,31 @@ export class TransactionOverviewView extends React.Component<
     });
   };
 
+  public redirectToTransactionType() {
+    const {
+      urlParams,
+      serviceTransactionTypes,
+      location,
+      history
+    } = this.props;
+    const { serviceName, transactionType } = urlParams;
+    const firstTransactionType = first(serviceTransactionTypes);
+    if (!transactionType && firstTransactionType) {
+      history.replace({
+        ...location,
+        pathname: `/${serviceName}/transactions/${firstTransactionType}`
+      });
+    }
+  }
+
+  public componentDidMount() {
+    this.redirectToTransactionType();
+  }
+
+  public componentDidUpdate() {
+    this.redirectToTransactionType();
```

Comment:
If the component mounts (`componentDidMount `), and the list of services has not been loaded yet, it won't redirect. When the services are loaded the component will re-render, this method will be called it will redirect to the first transaction type.